pkg/cvo/metrics.go

```diff
@@ -156,21 +212,50 @@ func RunMetrics(runContext context.Context, shutdownContext context.Context, lis
 		} else {
 			select {
 			case <-runContext.Done(): // clean shutdown
-			case result := <-resultChannel: // crashed before a shutdown was requested
+			case result := <-resultChannel: // crashed before a shutdown was requested or metrics server recreated
+				if restartServer {
+					klog.Info("Creating metrics server with updated TLS configutation.")
+					server = createHttpServer()
+					go startListening(server, tlsConfig, listenAddress, resultChannel)
+					restartServer = false
+					continue
+				}
 				resultChannelCount--
 				if result.error == nil {
 					klog.Infof("Collected metrics %s goroutine.", result.name)
 				} else {
 					klog.Errorf("Collected metrics %s goroutine: %v", result.name, result.error)
 					loopError = result.error
 				}
+			case event := <-watcher.Events:
+				if event.Op != fsnotify.Chmod && event.Op != fsnotify.Remove &&
+					certsChanged(origCertChecksum, origKeyChecksum, certFile, keyFile) {
+
+					// Update file checksums with latest files.
+					origCertChecksum = checksumFile(certFile)
+					origKeyChecksum = checksumFile(keyFile)
+
+					var tlsErr error
+					tlsConfig, tlsErr = makeTLSConfig(certFile, keyFile)
+					if tlsErr == nil {
+						restartServer = true
+						shutdownHttpServer(shutdownContext, server)
+						continue
+					} else {
+						loopError = fmt.Errorf("Failed to create TLS config: %w", err)
```

Comment:
"create" -> "reload" or something so the error string makes it more clear that we had been find, but then someone borked the key/cert files?

Reply:
I updated the error to clarify it's an updated TLS config. Also, there is this log entry that will always precede: "Certificate and key changed. Will recreate metrics server with updated TLS configuration."

Reply:
In 0e6136c727b05509c, this is `Failed to create TLS configuration with updated configuration`, which looks good to me.